pkg/pilot/cassandra/v3/pilot.go

```diff
@@ -39,9 +45,65 @@ func NewPilot(opts *PilotOptions) (*Pilot, error) {
 		nodeTool:            opts.nodeTool,
 	}
 
+	// hack to test the seedprovider works in k8s, this should be moved somewhere better
+	cfgPath := "/etc/cassandra/cassandra.yaml"
+	read, err := ioutil.ReadFile(cfgPath)
+	if err != nil {
+		return nil, err
+	}
+
+	newContents := strings.Replace(string(read),
+		"org.apache.cassandra.locator.SimpleSeedProvider",
+		"io.k8s.cassandra.KubernetesSeedProvider", -1)
+
+	newContents = strings.Replace(string(newContents),
+		"SimpleSnitch",
+		"GossipingPropertyFileSnitch", -1)
+
+	err = ioutil.WriteFile(cfgPath, []byte(newContents), 0)
+	if err != nil {
+		return nil, err
+	}
+
+	snitchSettings, err := getSnitchSettings(opts.kubeClientset)
+	if err != nil {
+		return nil, err
+	}
+
+	glog.V(4).Info("generated snitch:")
+	glog.V(4).Info(snitchSettings)
+
+	err = ioutil.WriteFile("/cassandra-rackdc.properties", []byte(snitchSettings), 0)
+	if err != nil {
+		return nil, err
+	}
+
 	return p, nil
 }
 
+func getSnitchSettings(clientset kubernetes.Interface) (string, error) {
+	nodeClientset := clientset.CoreV1().Nodes()
+
+	nodeName := os.Getenv("NODE_NAME")
+	node, err := nodeClientset.Get(nodeName, metav1.GetOptions{})
```

Comment:
So making this call is something that's blocked a similar PR on the ES pilot. I'm not sure if we want to allow Pilots to `get nodes`. I'll need to dig out the literature, but exposing node information to pods poses new security considerations as the pod is able to infer more information about where it is running.

I'd be happy to run with this for now until we hear more, but nodes are non-namespaced resources etc. so this is a more considerable change than first meets the eye. That said, I think we'd need some gRPC/synchronous (and highly available) distribution point for this configuration otherwise, which doesn't seem like the *first* thing we should rush to try.

Reply:
It'd be interesting to see those security worries. This sort of information about node placement is probably going to be necessary for most dbs, so maybe a minimal set of it could be located in the navigator api?

Reply:
Yep this is possible, but I'd rather keep it as-is for now. Copying that information across to Navigator could become more nuanced than we initially think - e.g. what if stale data exists on the Pilot resource, or if the data doesn't already exist?

I too am not certain on those security worries - let's not address it until we need to/we've got feedback from users 😄

Reply:
Another option to avoid giving access to `get nodes` is to send this information via labels and the downward api: We could watch for pilot pods and label them with the appropriate failure zones from the node once they're scheduled. This could then be exposed in the pod using `fieldPath: metadata.labels` from the downward api.

We'd have to wait for those labels to get propagated into the pod before starting cassandra, so it might cause too much latency when starting a cluster.